core/src/data/interval/primitive.rs

```diff
@@ -199,4 +199,15 @@ mod tests {
         assert_eq!(Month(8) / 4.0, Month(2));
         assert_eq!(8.0 / Month(4), Month(2));
     }
+
+    #[test]
+    fn arithmetic_microsecond() {
+        use Interval::*;
+
+        assert_eq!(Microsecond(2) * 3_i8, Microsecond(6));
+        assert_eq!(2_i8 * Microsecond(3), Microsecond(6));
+
+        assert_eq!(Microsecond(2) * 3_f64, Microsecond(6));
+        assert_eq!(2_f64 * Microsecond(3), Microsecond(6));
+    }
```

Comment:
This does not look related to this issue. Code itself looks nice, thanks :)
You can make the separate PR for this, then I'll approve right after.